core/pkg/runtime/from_config.go

```diff
@@ -126,6 +129,24 @@ func FromConfig(logger *logger.Logger, config Config) (*Runtime, error) {
 	}, nil
 }
 
+// buildTelemetryRecorder is a helper to build telemetry.MetricsRecorder based on configurations
+func buildTelemetryRecorder(logger *logger.Logger, config telemetry.Config) (*telemetry.MetricsRecorder, error) {
+	// Build metric reader based on configurations
+	mReader, err := telemetry.BuildMetricReader(context.TODO(), config)
+	if err != nil {
+		logger.Error(fmt.Sprintf("failed to setup metric reader: %v", err))
```

Comment:
since any error that occurs during the setup will by logged in [start.go](https://github.com/open-feature/flagd/blob/c9aba7ffd226c8eca363209528f56fe6dc575740/flagd/cmd/start.go#L163), I would rather return a wrapped error here:

```
return nil, fmt.Errorf("failed to setup metric reader: %w", err)
```

Reply:
Good suggestion. I improve error handling in this function based on the feedback :) 

Will update with the next commit